### PR TITLE
change the way iconography scss is generated

### PR DESCRIPTION
**Pull request by vxsx**
- mixin based icon approach
- renamed all the svgs like this `s/icon-(.*)/$1/` so 
the mixin could be called like `@include icon(something)` instead of `@include icon(icon-something)`

### CODE REVIEW

File: cms/static/cms/sass/components/_iconography.scss [hunk 1]
```diff
@@ -1,82 +1,158 @@
...
 // default font file generated by gulp
 @font-face {
     font-family: "django-cms-iconfont";
-    src: url("../fonts/django-cms-iconfont.eot?v=3.2.0");
-    src: url("../fonts/django-cms-iconfont.eot?#iefix&v=3.2.0") format("embedded-opentype"),
-         url("../fonts/django-cms-iconfont.woff?v=3.2.0") format("woff"),
-         url("../fonts/django-cms-iconfont.ttf?v=3.2.0") format("truetype"),
-         url("../fonts/django-cms-iconfont.svg?v=3.2.0#fontawesomeregular") format("svg");
-    font-weight: normal;
+    src: url('../fonts/django-cms-iconfont.eot');
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_iconography.scss]: except for the missing version, all good

File: cms/static/cms/sass/components/_iconography.scss [hunk 2]
```diff
@@ -5,73 +5,153 @@
...
 }
 
-.cms-icon {
+%icon {
     display: inline-block;
     font: normal normal normal 14px/1 django-cms-iconfont;
     font-size: inherit;
     text-rendering: auto;
+    transform: translate(0, 0);
     -webkit-font-smoothing: antialiased;
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_iconography.scss]: do we actually need these prefixes now?

File: cms/static/cms/sass/libs/_iconfont.scss [hunk 3]
```diff
@@ -3,31 +3,51 @@
...
 }
 
-.cms-icon {
+%icon {
     display: inline-block;
     font: normal normal normal 14px/1 <%= fontName %>;
     font-size: inherit;
     text-rendering: auto;
+    transform: translate(0, 0);
     -webkit-font-smoothing: antialiased;
```
Comment (FinalAngel) [inline cms/static/cms/sass/libs/_iconfont.scss]: we would have to change the prefix here

**Comment by FinalAngel** [comment]
LGTM

File: cms/static/cms/sass/components/_iconography.scss (hunk 2 above)
Comment (vxsx) [inline cms/static/cms/sass/components/_iconography.scss]: yes we still do since autoprefixer doesn't handle non-standard properties

**Pull request merged**